libp2p/network/swarm.py

```diff
@@ -173,11 +227,22 @@ def __init__(
         # Load balancing state
         self._round_robin_index = {}
 
+        # Start health monitoring if enabled
+        if self.connection_config.enable_health_monitoring:
+            self._start_health_monitoring()
+
     async def run(self) -> None:
         async with trio.open_nursery() as nursery:
             # Create a nursery for listener tasks.
             self.listener_nursery = nursery
             self.event_listener_nursery_created.set()
+
+            # Start health monitoring if enabled
+            if self.connection_config.enable_health_monitoring:
+                self._health_monitoring_task = nursery.start_soon(
```

Comment:
The return type of `nursery.start_soon` is `None`, so the check below doesn't really do anything.

Reply:
@pacrob Thanks for catching that! You're absolutely right - `nursery.start_soon()` returns `None`, so the check was useless. I've fixed it by replacing `_health_monitoring_task` with a boolean flag `_health_monitoring_active` that properly tracks the health monitoring state. The health monitoring task cancellation is now handled by the nursery cleanup automatically.